dartagnan/src/main/java/com/dat3m/dartagnan/expression/integers/IntCmpExpr.java

```diff
@@ -0,0 +1,21 @@
+package com.dat3m.dartagnan.expression.integers;
+
+import com.dat3m.dartagnan.expression.Expression;
+import com.dat3m.dartagnan.expression.ExpressionVisitor;
+import com.dat3m.dartagnan.expression.base.BinaryExpressionBase;
+import com.dat3m.dartagnan.expression.misc.CmpOp;
+import com.dat3m.dartagnan.expression.type.BooleanType;
+import com.dat3m.dartagnan.expression.utils.ExpressionHelper;
+
+public class IntCmpExpr extends BinaryExpressionBase<BooleanType, CmpOp> {
+
+    public IntCmpExpr(BooleanType type, CmpOp kind, Expression left, Expression right) {
+        super(type, kind, left, right);
+        ExpressionHelper.checkSameType(left, right);
```

Comment:
Yeah, will do.

Reply:
Meh, we also use this class to make equalities between boolean values.
Not sure what the best solution to this problem is:
- One could generalize the expression class since `==` and `!=` make sense for all types. But then we need to add checks that make sure the operator makes sense for the type (i.e. `==` on structs is fine, but `<` is not).
- Alternatively, one could separately add `==` to booleans (or rather `<=>` as `BoolBinaryOp`). But then we would not be able to make equalities between arbitrary types.
- One could make separate expressions for equalities and the other relational operators.
- Or something else...

Reply:
I added `BoolBinaryOp.IFF` and now (dis)equalities between booleans are encoded with that operator.